jobrunner/executors/local.py

```diff
@@ -533,6 +537,22 @@ def get_job_metadata(
     return job_metadata
 
 
+# keys that may be missing in older metadata files, and their default empty values
+METADATA_DEFAULTS = {
+    "hint": None,
+    "unmatched_patterns": [],
+    "unmatched_outputs": [],
```

Comment:
Is this going to bit us in the bum like mutable default arguments?

That is, if we end up using these lists because the metadata for a job is missing one of these keys, is there any risk that we'll mutate the list for one job and then accidentally mutate the defaults too?

Reply:
Hmm, it could do. Empty tuple, so its always iterable?